controller/script/simulate-proxy/main.go

```diff
@@ -244,109 +360,69 @@ func main() {
-					SourceIp: sourceIp,
-					Connects: count,
-					Disconnects: []*pb.TransportSummary{
-						&pb.TransportSummary{
-							DurationMs: uint64(randomCount()),
-							BytesSent:  uint64(randomCount()),
-						},
-					},
-					Protocol: common.Protocol_TCP,
-				},
-			},
-			Proxy: pb.ReportRequest_INBOUND,
-			Requests: []*pb.RequestScope{
-
-				// gRPC
-				&pb.RequestScope{
-					Ctx: &pb.RequestCtx{
-						SourceIp: sourceIp,
-						TargetAddr: &common.TcpAddress{
-							Ip:   targetIp,
-							Port: randomPort(),
-						},
-						Authority: "world.greeting:7778",
-					},
-					Count: count,
-					Responses: []*pb.ResponseScope{
-						&pb.ResponseScope{
-							Ctx: &pb.ResponseCtx{
-								HttpStatusCode: http.StatusOK,
-							},
-							ResponseLatencyCounts: randomLatencies(count),
-							Ends: randomGrpcEos(count),
-						},
-					},
-				},
-
-				// HTTP/2
-				&pb.RequestScope{
-					Ctx: &pb.RequestCtx{
-						SourceIp: sourceIp,
-						TargetAddr: &common.TcpAddress{
-							Ip:   targetIp,
-							Port: randomPort(),
-						},
-						Authority: "world.greeting:7778",
-					},
-					Count: count,
-					Responses: []*pb.ResponseScope{
-						&pb.ResponseScope{
-							Ctx: &pb.ResponseCtx{
-								HttpStatusCode: randomHttpResponseCode(),
-							},
-							ResponseLatencyCounts: randomLatencies(count),
-							Ends: randomH2Eos(count),
-						},
-					},
-				},
-			},
-
-			HistogramBucketBoundsTenthMs: latencyBucketBounds[:],
-		}
-
-		_, err = client.Report(context.Background(), req)
-		if err != nil {
-			log.Fatal(err.Error())
-		}
-
-		time.Sleep(*sleep)
+	stopCh := make(chan os.Signal)
+	signal.Notify(stopCh, os.Interrupt, os.Kill)
+
+	for _, addr := range strings.Split(*metricsAddrs, ",") {
+		randomNamespace := getRandomNamespaceKey(podsByNamespace)
+		proxyPod := podList[rand.Intn(len(podList))]
+
+		go func(address string, namespace string, pod *v1.Pod) {
+
+			proxy := simulatedProxy{
+				sleep:     *sleep,
+				pod:       pod,
+				namespace: namespace,
+				proxyMetricCollectors: &proxyMetricCollectors{requestTotals: prom.NewCounterVec(
+					prom.CounterOpts{
+						Name: "request_total",
+						Help: "A counter of the number of requests the proxy has received",
+					}, labels),
+					responseTotals: prom.NewCounterVec(
+						prom.CounterOpts{
+							Name: "response_total",
+							Help: "A counter of the number of responses the proxy has received.",
+						}, labels),
+					requestDurationMs: prom.NewHistogramVec(
+						prom.HistogramOpts{
+							Name:    "request_duration_ms",
+							Help:    "A histogram of the duration of a response",
+							Buckets: latencyBucketBounds,
+						}, labels),
+					responseLatencyMs: prom.NewHistogramVec(
+						prom.HistogramOpts{
+							Name:    "response_latency_ms",
+							Help:    "A histogram of the total latency of a response.",
+							Buckets: latencyBucketBounds,
+						}, labels)},
+			}
+
+			registerer := prom.NewRegistry()
```

Comment:
nice!

consider moving this whole block, starting with `proxy := simulatedProxy{` down to and including registerer.MustRegister, into something like:

```go
func NewSimulatedProxy(address string, namespace string, pod *v1.Pod) *simulatedProxy {...}
```